static/components.interfaces.ts

```diff
@@ -329,3 +336,130 @@ export type PopulatedDeviceViewState = StateWithId & {
+ * states. Components can have either empty (default) or populated states.
+ */
+export interface ComponentStateMap {
+    [COMPILER_COMPONENT_NAME]: EmptyCompilerState | PopulatedCompilerState | CompilerForTreeState;
+    [EXECUTOR_COMPONENT_NAME]: EmptyExecutorState | PopulatedExecutorState | ExecutorForTreeState;
+    [EDITOR_COMPONENT_NAME]: EmptyEditorState | PopulatedEditorState;
+    [TREE_COMPONENT_NAME]: EmptyTreeState;
+    [OUTPUT_COMPONENT_NAME]: OutputState;
+    [TOOL_COMPONENT_NAME]: ToolViewState;
+    [TOOL_INPUT_VIEW_COMPONENT_NAME]: EmptyToolInputViewState | PopulatedToolInputViewState;
+    [DIFF_VIEW_COMPONENT_NAME]: EmptyDiffViewState | PopulatedDiffViewState;
+    [OPT_VIEW_COMPONENT_NAME]: EmptyOptViewState | PopulatedOptViewState;
+    [STACK_USAGE_VIEW_COMPONENT_NAME]: EmptyStackUsageViewState | PopulatedStackUsageViewState;
+    [FLAGS_VIEW_COMPONENT_NAME]: EmptyFlagsViewState | PopulatedFlagsViewState;
+    [PP_VIEW_COMPONENT_NAME]: EmptyPpViewState | PopulatedPpViewState;
+    [AST_VIEW_COMPONENT_NAME]: EmptyAstViewState | PopulatedAstViewState;
+    [GCC_DUMP_VIEW_COMPONENT_NAME]: EmptyGccDumpViewState | PopulatedGccDumpViewState;
+    [CFG_VIEW_COMPONENT_NAME]: EmptyCfgViewState | PopulatedCfgViewState;
+    [CONFORMANCE_VIEW_COMPONENT_NAME]: PopulatedConformanceViewState;
+    [IR_VIEW_COMPONENT_NAME]: EmptyIrViewState | PopulatedIrViewState;
+    [CLANGIR_VIEW_COMPONENT_NAME]: EmptyClangirViewState | PopulatedClangirViewState;
+    [OPT_PIPELINE_VIEW_COMPONENT_NAME]: EmptyOptPipelineViewState | PopulatedOptPipelineViewState;
+    [LLVM_OPT_PIPELINE_VIEW_COMPONENT_NAME]: EmptyOptPipelineViewState | PopulatedOptPipelineViewState;
+    [RUST_MIR_VIEW_COMPONENT_NAME]: EmptyRustMirViewState | PopulatedRustMirViewState;
+    [HASKELL_CORE_VIEW_COMPONENT_NAME]: EmptyHaskellCoreViewState | PopulatedHaskellCoreViewState;
+    [HASKELL_STG_VIEW_COMPONENT_NAME]: EmptyHaskellStgViewState | PopulatedHaskellStgViewState;
+    [HASKELL_CMM_VIEW_COMPONENT_NAME]: EmptyHaskellCmmViewState | PopulatedHaskellCmmViewState;
+    [GNAT_DEBUG_TREE_VIEW_COMPONENT_NAME]: EmptyGnatDebugTreeViewState | PopulatedGnatDebugTreeViewState;
+    [GNAT_DEBUG_VIEW_COMPONENT_NAME]: EmptyGnatDebugViewState | PopulatedGnatDebugViewState;
+    [RUST_MACRO_EXP_VIEW_COMPONENT_NAME]: EmptyRustMacroExpViewState | PopulatedRustMacroExpViewState;
+    [RUST_HIR_VIEW_COMPONENT_NAME]: EmptyRustHirViewState | PopulatedRustHirViewState;
+    [DEVICE_VIEW_COMPONENT_NAME]: EmptyDeviceViewState | PopulatedDeviceViewState;
+}
+
+/**
+ * Type-safe component configuration that enforces:
+ * - type must be the literal string 'component' (not just any string)
+ * - componentName must be a valid component name from ComponentStateMap
+ * - componentState must match the expected type for that component
+ */
+export interface ComponentConfig<K extends keyof ComponentStateMap = keyof ComponentStateMap> {
```

Comment:
It seems wrong to let this have a default - you would want to specify a specific pane if I'm not mistaken?

If we'd want to refer to any component config, it might be better to do

```suggestion
export type AnyComponentConfig = ComponentConfig<keyof ComponentStateMap>
export interface ComponentConfig<K extends keyof ComponentStateMap> {
```